dissect/target/plugins/os/windows/regf/usb.py

```diff
@@ -35,127 +44,173 @@
+                    }
+
+                    values = [v.name for v in usb_device.values()]
+                    if "FriendlyName" in values:
+                        friendly_name = usb_device.value("FriendlyName").value
+
+                    if "ContainerID" in values:
+                        container_id = usb_device.value("ContainerID").value
+
                     try:
-                        friendlyname = usb_device.value("FriendlyName").value
-                        # NOTE: make this more gracefull, windows 10 does not have the LogConf subkey
-                        timestamps = (
-                            self.unpack_timestamps(properties[2])
-                            if len(properties) == 3
-                            else self.unpack_timestamps(properties[1])
-                        )
-                        # ContainerIDs can be found back in USB and WdpBusEnumRoot
-                        containerid = usb_device.value("ContainerID").value
-                    except RegistryValueNotFoundError:
-                        friendlyname = None
-                        timestamps = {
-                            "first_install": None,
-                            "first_insert": None,
-                            "last_insert": None,
-                            "last_removal": None,
-                        }
-                        containerid = None
-
-                    yield UsbRegistryRecord(
-                        device_type=device_info["device_type"],
-                        friendlyname=friendlyname,
+                        timestamps = unpack_timestamps(usb_device.subkey("Properties"))
+
+                    except RegistryValueNotFoundError as e:
+                        self.target.log.warning(f"Unable to parse USBSTOR registry properties for serial {serial}")
+                        self.target.log.debug("", exc_info=e)
+                        pass
+
+                    mounts = list(self.find_mounts(serial))
+
+                    yield USBRegistryRecord(
+                        friendly_name=friendly_name,
                         serial=serial,
-                        vid=None,
-                        pid=None,
-                        vendor=device_info["vendor"],
-                        product=device_info["product"],
-                        version=device_info["version"],
-                        containerid=containerid,
-                        first_install=timestamps["first_install"],
-                        first_insert=timestamps["first_insert"],
-                        last_insert=timestamps["last_insert"],  # AKA first arrival
-                        last_removal=timestamps["last_removal"],
-                        info_origin=info_origin,
+                        container_id=container_id,
+                        **device_info,
+                        **timestamps,
+                        volumes=list(self.find_volumes(serial)),
+                        mounts=mounts,
```

Comment:
```suggestion
                        mounts=list(self.find_mounts(serial)),
```

Reply:
This one is also still relevant. This is the same as volumes, either do both outside the kwarg or inside.

Reply:
> We need the result of `self.find_mounts(serial)` for us to calculate `users` so I think this cannot be moved to the `UsbRegistryRecord` initialization. We can move `volumes` up to `mounts = ` if you want.

Oops, missed that initially, sorry!